accession/accession.py

```diff
@@ -75,21 +75,62 @@ def get_step_run_id(self, encode_file):
             step_run_id = step_run.get("@id")
         return step_run_id
 
-    def file_at_portal(self, file):
+    def get_encode_file_matching_md5_of_blob(self, file):
+        """Finds an ENCODE File object whose md5sum matches md5 of a blob in URI in backend.
+
+        Args:
+            file (str): String representing an URI to an object in the backend.
+
+        Returns:
+            dict: Dictionary representation of the matching file object on portal.
+            None if no matching objects are found.
+        """
         md5sum = self.backend.md5sum(file)
         search_param = [("md5sum", md5sum), ("type", "File")]
-        encode_file = self.conn.search(search_param)
-        if len(encode_file) > 0:
-            return self.conn.get(encode_file[0].get("@id"))
+        encode_files = self.conn.search(search_param)
+        filtered_encode_files = type(self).filter_encode_files_by_status(encode_files)
+        if filtered_encode_files:
+            if len(filtered_encode_files) > 1:
+                self.logger.warning(
+                    "get_encode_file_matching_md5_of_blob found more than 1 files matching the md5 of the blob."
+                )
+            return self.conn.get(filtered_encode_files[0].get("@id"))
+        else:
+            return None
+
+    @staticmethod
+    def filter_encode_files_by_status(
+        encode_files, forbidden_statuses=["replaced", "revoked", "deleted"]
```

Comment:
Should be a tuple (try to avoid mutable default arguments):
```suggestion
        encode_files, forbidden_statuses=("replaced", "revoked", "deleted")
```

Reply:
fNot really an issue here because the `forbidden_statuses` is not being changed anywhere in the function.